cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/PullCommand.java

```diff
@@ -175,28 +168,14 @@ public void execute() {
                     "unexpected error occurred while creating package repository in bala cache: " + e.getMessage());
         }
 
-        for (String supportedPlatform : SUPPORTED_PLATFORMS) {
-            try {
-                Settings settings;
-                try {
-                    settings = readSettings();
-                    // Ignore Settings.toml diagnostics in the pull command
-                } catch (SettingsTomlException e) {
-                    // Ignore 'Settings.toml' parsing errors and return empty Settings object
-                    settings = Settings.from();
-                }
-                CentralAPIClient client = new CentralAPIClient(RepoUtils.getRemoteRepoURL(),
-                                                               initializeProxy(settings.getProxy()),
-                                                               getAccessTokenOfCLI(settings));
-                client.pullPackage(orgName, packageName, version, packagePathInBalaCache, supportedPlatform,
-                                   RepoUtils.getBallerinaVersion(), false);
-            } catch (PackageAlreadyExistsException e) {
-                errStream.println(e.getMessage());
-                CommandUtil.exitError(this.exitWhenFinish);
-            } catch (CentralClientException e) {
-                errStream.println("unexpected error occurred while pulling package:" + e.getMessage());
-                CommandUtil.exitError(this.exitWhenFinish);
-            }
+        try {
+            ProjectCentralUtils.pullPackage(orgName, packageName, version);
```

Comment:
If we change `pullPackage` implememntation, we should check all the scenarios of `pull` command.